### ISSUE TITLE
lightline status line event hooks not working

### ISSUE THREAD

**Issue by bitwombat** [ludovicchabant/vim-gutentags#240]
**Describe the bug**
the 'ctags' text on the status line isn't going way.

Here's my lightline setup:

```
let g:lightline = {
    \ 'colorscheme': 'bitwombat',
    \ 'active': {
    \   'left': [ [ 'mode', 'paste' ],
    \             [ 'filename', 'readonly', 'modified' ],
    \             [ 'gitbranch' ],
    \           ],
    \   'right': [
    \             [ 'percent' ],
    \             [ 'lineinfo' ],
    \             [ 'fileformat', 'fileencoding' ],
    \             [ 'gutentags'],
    \            ],
    \ },
    \ 'component_function': {
    \   'readonly': 'LightlineReadonly',
    \   'filename': 'LightlineFilename',
    \   'gitbranch': 'fugitive#head',
    \   'fileformat': 'LightlineFileformat',
    \   'fileencoding': 'LightlineFileencoding',
    \   'gutentags': 'gutentags#statusline',
    \ },
    \ }

```
Then I have the recommended autocommands:
```
augroup MyGutentagsStatusLineRefresher
            autocmd!
            autocmd User GutentagsUpdating call lightline#update()
            autocmd User GutentagsUpdated call lightline#update()
augroup END
```

**Share your setup**
- Ubuntu 16.04 and Vim 7.4.1689
- Universal Ctags 0.0.0(d080788), based on Exuberant Ctags 5.8
- Are you using `g:gutentags_cache_dir`?
no

**Post the logs**
https://gist.github.com/bitwombat/e8726dbb7fb671ab314ce6f6966f3327

**Comment by bitwombat** [comment]
Update: I can tell the event's actually firing, because if I put something wrong in the cmd, I get an error in vim.
Also, if anything else updates the status line, like just moving the cursor down a line, it updates.

So not a big problem, and probably lightline's problem? Any input appreciated anyway.

**Comment by ludovicchabant** [comment]
Before Vim8, there was not good way of knowing if the background script was done updating tags... so it was effectively a "pull model". We were checking if the script was finished any time the status line updated, so yeah, you'd have to move the cursor or type something for the status line icon to disappear.

With Vim8/Neovim, we now have support for proper background jobs so the `GutentagsUpdated` command will fire asynchronously, as soon as it's done, and that should update the status line as expected...

Basically, this is a long way of saying: "it was a hack in Vim7, if you update to Vim8, it should work better".
